File: lib/mylive_config.dart
class MyLiveVideoConfig {
  /// bps
  int bitrate = 1000 * 1000;

  /// width e.g. 640, 854, 1280, 1920, 2560, 3840
  int width = 1280;

  /// height e.g. 360, 480, 720, 1080, 1440, 2160
  int height = 720;

  /// fps e.g. 30, 60
  int fps = 30;

  MyLiveVideoConfig({this.bitrate = 1000 * 1000, this.width = 1280, this.height = 720, this.fps = 30});
}

class MyLiveAudioConfig {
  /// The video bitrate in bps
  int bitrate = 128 * 1000;

  /// sample rate
  int sampleRate = 44100;

  MyLiveAudioConfig({this.bitrate = 128 * 1000, this.sampleRate = 44100});
}

File: ios/Classes/MyLiveConfig.swift
import Foundation
import Network

public struct MyLiveAudioConfig {
    public let bitrate: Int
    public init(bitrate: Int = 128_000) {
        self.bitrate = bitrate
    }
}

public struct MyLiveVideoConfig {
    public let bitrate: Int
    public let width: Int
    public let height: Int
    public let fps: Float64
    public let gopDuration: TimeInterval

    public init(
        fps: Float64 = 30,
        gopDuration: TimeInterval = 1.0
    ) {
        self.bitrate = 1_000_000
        self.fps = fps
        self.gopDuration = gopDuration
        self.width = 1280
        self.height = 720
    }

    public init(
        bitrate: Int,
        width: Int = 1280,
        height: Int = 720,
        fps: Float64 = 30,
        gopDuration: TimeInterval = 1.0
    ) {
        self.bitrate = bitrate
        self.fps = fps
        self.width = width
        self.height = height
        self.gopDuration = gopDuration
    }
}

enum ConfigurationError: Error {
    case invalidParameter(String)
}
